src/msg/consumer/server.go

```diff
@@ -62,3 +69,43 @@ func (h *handler) Handle(conn net.Conn) {
 }
 
 func (h *handler) Close() {}
+
+type messageHandler struct {
```

Comment:
Would it be simpler to make this public so you don't have to change `serverOptions` and `NewServer`?

Reply:
I could make this public, but I'd still keep the changes in `NewServer`, seems easier to use that way. Users just need to implement `ConsumeFn` or `MessageFn` without needing to know what kind of handler to create.

Reply:
Plus the handler itself is meaningless without the context of `server.Server`

Reply:
I see, I was just thinking the following might be simpler:

```go
func (h *MessageHandler) Consume(c consumer.Consumer) {
	var (
		msgErr error
		msg    Message
	)
	for {
		msg, msgErr = c.Message()
		if msgErr != nil {
			break
		}
		h.messageFn(msg)
	}
	if msgErr != nil && msgErr != io.EOF {
		h.opts.InstrumentOptions().Logger().WithFields(log.NewErrField(msgErr)).Errorf("could not read message from consumer")
	}
	c.Close()
}
```

Then to use it one could do:

```go
msgHandler := NewMessageHandler(msgFn, handlerOpts)
opts := NewServerOptions().SetConsumeFn(msgHandler.Consume)
server := NewServer(addr, opts)
```

and you wouldn't need to update any of the code any place else.

Reply:
Yeah that's a good idea, I did some refactoring based on the feedback